### PR TITLE
Updating logical definitions for compatibility with RO domains and ranges

### PR DESCRIPTION
**Pull request by balhoff**
These changes fix 224 unsatisfiable classes which are found when HPO is used with RO. There are quite a few remaining, but many of these are due to issues in Uberon.

### CODE REVIEW

**Comment by balhoff** [comment]
@pnrobinson my apologies, I know you don't really want pull requests, but I didn't want to make all these changes without running it by you first.

I think we should update the RO import file to include more axioms, to prevent these incompatibilities. But the existing conflicts need to be fixed first.

**Comment by pnrobinson** [comment]
THanks, Jim. I am certainly not against PRs, but just think it is over-engineered for most of our work with hp edit. I will review this shortly.

**Comment by drseb** [comment]
@balhoff a reason to avoid PRs was also that they became unmergeable when other commits are getting in between, due to Protege's way of writing the owl-files (i.e. line-ordering). If it is save now with protege, you would have my vote to create more PRs.

**Comment by balhoff** [comment]
@drseb I think the line ordering is pretty stable now.

One challenge with a pull request like this is if someone wants to actually review, it's quite hard to visually inspect the changes. I tried to generate a diff using ROBOT, but it wasn't any easier to read.

**Comment by pnrobinson** [comment]
The utility of the pull request is put into question by the formating and difficulties of really understanding what it being requested. This is why it might be easier to simply discuss thiese things as issues and then to LOCK and implement. Even if it is clunky...

**Comment by balhoff** [comment]
@pnrobinson @drseb @cmungall I made a quick tool. Here's a diff for this pull request:

https://stars-app.renci.org/owldiff/diff?right=https://github.com/obophenotype/human-phenotype-ontology/raw/eb792f57e60a1a8209834048ca326e55f8fbaa4a/src/ontology/hp-edit.owl&left=https://github.com/obophenotype/human-phenotype-ontology/raw/master/src/ontology/hp-edit.owl&loadimports=true

**Comment by drseb** [comment]
Very cool! Can you force it to always list "removed" first and "added" second?

**Comment by balhoff** [comment]
@drseb I updated the formatting to group removed first, then added.

I do see that I triggered several declaration axioms for Uberon terms. I can remove these if it seems important.

**Comment by balhoff** [comment]
One caution - it doesn't seem to be rendering axiom annotations: https://github.com/balhoff/owl-diff/issues/1

(not relevant for this pull request)

**Comment by drseb** [comment]
Getting back to this PR; saw this line:

```
Ascites - http://purl.obolibrary.org/obo/HP_0001541
Removed

Ascites EquivalentTo has part some (increased amount and (inheres in some (interstitial fluid and (part of some peritoneal cavity))) and (has modifier some abnormal))
Added

Ascites EquivalentTo has part some (increased amount and (inheres in some (interstitial fluid and (RO_0001025 some peritoneal cavity))) and (has modifier some abnormal))

```

RO_0001025 is located_in. Why isn't the label shown. Is it missing from RO_import?

**Comment by drseb** [comment]
Have you checked that this complex def leads to correct inferences? 
`Gout EquivalentTo has part some (increased amount and (inheres in some (uric acid and (part of some blood))) and (has modifier some abnormal) and (RO_0003302 some (increased amount and (inheres in some (arthritis and (occurs in some skeletal joint))))))`

**Comment by balhoff** [comment]
Yes, RO_import seems to be missing 'located in'. What's the procedure for adding terms there? The other term that's needed (related to your 'Gout' question is:

'causes or contributes to condition' : RO_0003302

**Comment by balhoff** [comment]
Re: Gout
> Have you checked that this complex def leads to correct inferences?

There are no additional inferred superclasses besides the two asserted ones:
- Arthritis
- Hyperuricemia

This one needed to change because 'causally upstream of or within' should be occurrent -> occurrent, while this definition is relating 'dependent continuant' -> 'dependent continuant'. It is a complex definition and I wanted to correct this one problem (and placement of 'occurs in') without getting into how else it might be constructed. It might be good for @cmungall to take a look at the use of 'causes or contributes to condition'.

**Comment by drseb** [comment]
I updated ro_import.owl (let me know if I should show you how this is currently done). Your tool now shows 'located in' and 'causes or contributes to condition'. Fantastic. 

I think we can merge. Are we waiting for somebody else to approve?

**Comment by balhoff** [comment]
Sounds great to me. If anyone wants to look further at the Gout definition relationships, we can revise it separately. I think it is good for now.

**Comment by cmungall** [comment]
OK, so when this is combined with ro_import it causes unsats:

image

Not that this is fixed in the current uberon - so regenerating uberon_imports should fix

**Comment by pnrobinson** [comment]
?It seems to me that a pit is not a subclass of an anatomical surface. Shouldnt it be part-of?

-Peter


Peter Robinson

Professor and Donald A. Roux Chair, Genomics and Computational Biology

The Jackson Laboratory for Genomic Medicine

10 Discovery Drive

Farmington, CT 06032

860.837.2095 t | 860.990.3130 m

peter.robinson@jax.org<mailto:peter.robinson@jax.org>

www.jax.org

Robinson lab: https://robinsongroup.github.io/

New Book: Computational Exome and Genome Analysis<https://www.crcpress.com/Computational-Exome-and-Genome-Analysis/Robinson-Piro-Jager/p/book/9781498775984>

The Jackson Laboratory: Leading the search for tomorrow's cures

________________________________

**Comment by cmungall** [comment]
> ?It seems to me that a pit is not a subclass of an anatomical surface.

correct. That was the issue in the uberon ticket linked

> Shouldnt it be part-of?

The fix was to make it a type of surface region. These upper level classes are a bit squishy..

The overriding issue here is that the release manager has to flush the cached version of the full ontology before making the import module. This is all due for an overhaul

**Comment by pnrobinson** [comment]
Please do not make causally upstream of part of the definition. Note there is a disease entity gout but we are talking about gouty arthritis, the pheno feature! I thought the current definition was at least reasonable.

**Comment by balhoff** [comment]
Hi Peter, Gout changed from

`
Gout EquivalentTo has part some (increased amount and (inheres in some (uric acid and (part of some blood))) and (causally upstream of or within some (increased amount and (part of some skeletal joint) and (inheres in some arthritis))) and (has modifier some abnormal))
`

to

`
Gout EquivalentTo has part some (increased amount and (inheres in some (uric acid and (part of some blood))) and (has modifier some abnormal) and (causes or contributes to condition some (increased amount and (inheres in some (arthritis and (occurs in some skeletal joint))))))
`

I kept it pretty much the same, just replaced `causally upstream of or within` with `causes or contributes to condition` and fixed `part of` vs. `occurs in`.

So the causation part was already there. It seems like we need a separate discussion about this definition.

**Comment by pnrobinson** [comment]
This definition is wrong -- perhaps we also need to rename the term to gouty arthritis, which can be defined as Buildup of uric acid crystals in a joint associated with arthritis. 

I think that the following is closer, although it does not state that it is the same skeletal joint!

Gout EquivalentTo 
has part some (increased amount and (inheres in some (uric acid and (part of some skeletal joint))) 
and (has modifier some abnormal) 
and some (increased amount and (inheres in some (arthritis and (occurs in some skeletal joint))))
